### PR TITLE
Add sha384/sha512 digest function

### PR DESCRIPTION
**Pull request by hunshcn**
As the pre-content supporting sha384/sha512 remote_asset.

Related https://github.com/buildbuddy-io/buildbuddy/pull/6911

### CODE REVIEW

**Review by sluongng — CHANGES_REQUESTED** [review]
Could you please add some tests for these 2 hash functions in `digest_test.go`?

**Comment by hunshcn** [comment]
Generally sha1/sha512 is not used as a storage digest function. So I only added the download part, which should be enough.

**Comment by sluongng** [comment]
afaik, SHA384 is not used anywhere in the Bazel ecosystem. Do you have a use case for it?

**Comment by hunshcn** [comment]
> afaik, SHA384 is not used anywhere in the Bazel ecosystem. Do you have a use case for it?

yq/jq in aspect's bazel-lib.


https://github.com/search?q=%22integrity+%3D+%5C%22sha384%22+path%3A%22bzl%22&type=code
https://github.com/search?q=%22sha384-%22+path%3A*.bzl+NOT+is%3Afork&type=code

Not much indeed. Most of them are from https://github.com/bazel-contrib/rules-template/blob/a71e0a7624aa6e6fcde3d6623d3cb135fffbe28a/mylang/private/versions.bzl#L9

File: server/remote_cache/digest/digest_test.go [hunk 1]
```diff
@@ -58,11 +58,21 @@ func TestParseResourceName(t *testing.T) {
...
+		},
 		{ // download, resource with digest only
 			resourceName: "/blobs/072d9dd55aacaa829d7d1cc9ec8c4b5180ef49acac4a3c2f3ca16a3db134982d/1234",
 			matcher:      downloadRegex,
 			wantParsed:   newCASResourceName(&repb.Digest{Hash: "072d9dd55aacaa829d7d1cc9ec8c4b5180ef49acac4a3c2f3ca16a3db134982d", SizeBytes: 1234}, "", repb.DigestFunction_SHA256),
 		},
+		{ // download, resource with digest only
+			resourceName: "/blobs/b2d1d285b5199c85f988d03649c37e44fd3dde01e5d69c50fef90651962f48110e9340b60d49a479c4c0b53f5f07d690686dd87d2481937a512e8b85ee7c617f/1234",
+			matcher:      downloadRegex,
+			wantParsed:   newCASResourceName(&repb.Digest{Hash: "b2d1d285b5199c85f988d03649c37e44fd3dde01e5d69c50fef90651962f48110e9340b60d49a479c4c0b53f5f07d690686dd87d2481937a512e8b85ee7c617f", SizeBytes: 1234}, "", repb.DigestFunction_SHA512),
```
Comment (sluongng) [inline server/remote_cache/digest/digest_test.go:74]: Could you please add another test for SHA384 here?

**Review by sluongng — COMMENTED** [review]
Apart from the last minor nit. This PR looks good to me.

**Review by tylerwilliams — APPROVED** [review]

**Pull request merged**